rook/XMLDiff.py

```diff
@@ -335,17 +334,44 @@ def fail_message(*args):
         if len(a_element) > 0:
           a_string = ET.tostring(a_element)
           if len(a_string) > 80:
-            message.append('Branches in gold not matching test...\n{}'.format(path))
+            message.append(f'Branches in gold not matching test...\n{path}')
           else:
-            message.append('Branches in gold not matching test...\n{} {}'.format(path, a_string))
+            message.append(f'Branches in gold not matching test...\n{path} {a_string}')
         if len(b_element) > 0:
           b_string = ET.tostring(b_element)
           if len(b_string) > 80:
-            message.append('Branches in test not matching gold...\n{}'.format(path))
+            message.append(f'Branches in test not matching gold...\n{path}')
           else:
-            message.append('Branches in test not matching gold...\n{} {}'.format(path, b_string))
+            message.append(f'Branches in test not matching gold...\n{path} {b_string}')
   return (same, message)
 
+
+def ignore_subnodes_from_root(a_element, b_element, ignored_nodes):
+  """
+    Compares two element trees and returns (same,message) where same is true
+      if they are the same, and message is a list of the differences
+    @ In, a_element, ET.Element.Tree, the first element tree
+    @ In, b_element, ET.Element.Tree, the second element tree
+    @ In, ignored_nodes, str, address of ignored subnodes of XPath format, e.g.,
+                              `.//parentNode/childNode[@name:<>]/grandchildNode`
+    @ Out, a_element, ET.Element.Tree, the first element tree
+    @ Out, b_element, ET.Element.Tree, the second element tree
+    @ Out, success, bool, was the removal successful?
+  """
+  # internal check of ignored nodes already conducted by this point
+  for ignored in ignored_nodes:
+    try:
+      for element in [a_element, b_element]:
+        parentpath = '/'.join(ignored.split('/')[:-1]) # `.//parentNode/childNode[@name:<>]`
+        childpath = ignored.split('/')[-1]             # `/grandchildNode`
+        parentnode = element.find(parentpath)
+        parentnode.remove(parentnode.find(childpath))
+      success = True
+    except Exception as exp:
+      print((f"Could not find XPath {ignored} in elements with exception:{exp}"))
+      return a_element, b_element, False
```

Comment:
The return will happen if the node can not be found in the first element. In the situation when the node only contained in the second element, the return will still keep the ignored node the second element. I would suggest to remove ignored node in both elements first, and then return them.

Reply:
that's a good point, I didn't consider the case where we ignore a node found in only one of the element trees. 

I have removed the try-except for a simpler search. It now only exits out of the method when the ignored node is not found in _either_ of the element trees. 

I have also added a test for this case to `TestXMLDiffer.py`. Thanks for catching that!